lib/vdsm/API.py

```diff
@@ -1809,11 +1809,14 @@ class ManagedVolume(APIBase):
 
     @api.logged(on="api.storage")
     @api.method
-    def attach_volume(self, vol_id, connection_info):
+    def attach_volume(self, vol_id, connection_info, sd_id=None):
```

Comment:
If we use blank uuid in the schema, we can use here sd_id without default value, and pass it as positional argument to all APIs.